src/hardware/serialport/misc_util.cpp

```diff
@@ -148,49 +147,49 @@ bool TCPClientSocket::GetRemoteAddressString(Bit8u* buffer) {
 	return true;
 }
 
-bool TCPClientSocket::ReceiveArray(Bit8u* data, Bitu* size) {
-	if(SDLNet_CheckSockets(listensocketset,0))
-	{
+bool TCPClientSocket::ReceiveArray(uint8_t *data, uint32_t *size)
+{
+	if (SDLNet_CheckSockets(listensocketset, 0)) {
 		Bits retval = SDLNet_TCP_Recv(mysock, data, static_cast<int>(*size));
```

Comment:
`SDLNet_TCP_Recv` at most only fills `data` with up to `int` elements though [API link]. 
For my own understanding, is the concern here that `int` is too loosely defined by the standard? (ie: and is essentially implementation-dependent, so could in theory jump to 64 or 128bits willy-nilly on some architectures?).

In that case, yes - sounds like we need `size_t`'s whereever we tied into API's that deal with `int`.

Reply:
> `SDLNet_TCP_Recv` at most only fills `data` with up to `int` elements [[API link](http://sdl.beuc.net/sdl.wiki/SDLNet_TCP_Recv)], so I had though that `uint32` should cover it (2^32 > 2^31); but this is probably wrong?

`int` might be smaller, or bigger than 32-bits.

> For my own understanding, is the concern here that `int` is too loosely defined by the standard? (ie: and is essentially implementation-dependent, so could in theory jump to 64 or 128bits willy-nilly on some architectures?).

You are exactly right. `int` is (on purpose) defined only in relation to other types: sizeof(int) >= sizeof(short int) and sizeof(int) <= sizeof(long int) :/.

`size_t` is defined as type with minimum of 16-bits, that can hold the size of a maximum object stored in the memory (i.e. we don't know if it's even 32-bits, but we can be sure that it can store size of any buffer allocated on a given OS).

Modern CPUs are called 64-bits, but essentially they could be advertised as 512-bit CPUs as well (e.g. any CPU supporting AVX-512). I can't even say if the decision about int being 32-bit happens on hardware manufacturer side, on OS-developer side, or is it completely implementation-defined by C++ compiler vendors…

> In that case, yes - sounds like we need `size_t`'s whenever we tie into 3rd party code that uses loosely defined types like `int`. How do we `printf` `size_t`'s, given they're not pointers (PRIuPTR)s?

The correct way of printf'ing `size_t` is… `%zu` (btw, we should re-check if GCC on msys2 still has problem with %zu - it was the only compiler/env combination showing that issue).

Reply:
This helps a lot; thanks @dreamer.  I've moved to `size_t`'s for the array and index positions outside of SDL, and then cast down to `int` like you suggest when passed to SDL.  We'll see if `%zu` works (which I'm now using in a couple places), when it hits CI.